### PR TITLE
WEEK 4: Add AnalyticsRegistry smart contract with on-chain result tracking

### PR DESCRIPTION
**Pull request by santhil-cyber**
This adds the `AnalyticsRegistry` smart contract and wires up the backend stubs so the analytics service knows where to fetch registered results from.

### Smart Contract (`AnalyticsRegistry.sol`)
- Users can register analytics results on-chain, linking a **source dataset CID** → **result CID** with an analysis type
- Built-in **duplicate prevention** — same result CID can't be registered twice
- **Paginated history** via `getMyAnalytics(offset, limit)` so users can browse their past analyses without hitting gas limits
- `getAnalyticsForDataset()` returns all result CIDs linked to a given source dataset
- Events emitted on every registration for off-chain indexing

### Hardhat Test Suite (14 tests)
- Input validation (empty CID, empty analysis type)
- Duplicate result CID rejection
- Pagination edge cases (overflow, out-of-bounds offset)
- Wallet isolation (analyst A can't see analyst B's records)

### Bug Fixes
- **DocumentStorage tests** — fixed 4 failing tests that were passing raw strings to `bytes32` mappings (`documentPrices`, `documentOwners`). Now correctly computes `keccak256` before querying.

### Analytics Service Updates
- Added registry lookup stub endpoints: `GET /analytics/results/{result_cid}` and `GET /analytics/dataset/{dataset_cid}`
- Added `RegistryResultResponse` and `RegistryDatasetResponse` Pydantic schemas
- Updated README to reflect current project status

### Other
- Added `.env.example` with safe placeholder values for Sepolia RPC and deployer key
- Added demo script (`scripts/demoRegistry.js`) showing the full register → query flow
- Added Hardhat network config for Sepolia deployment

### CODE REVIEW

File: nextjs-frontend/lib/contractService.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { ethers, BrowserProvider, Contract } from 'ethers';
-import type { ContractABI } from './types';
+import type { ContractABI, AnalyticsRecord } from './types';
 
 const CONTRACT_ADDRESS = '0xd58de64aac08d5412b8020c7c61b215fec0c9644';
+const ANALYTICS_REGISTRY_ADDRESS = '0x9148Cd47B9c166CC651F57B8BfA77bf66496A90f';
```
Comment (karthiksathishjeemain) [inline nextjs-frontend/lib/contractService.ts]: Please don't make this address public. Retreive from a .env file

**Review by karthiksathishjeemain — COMMENTED** [review]
Have pointed out a small correction that is to be taken. Other part looks  great. Once it is done, let me know. Will approve then. Good work.

**Comment by santhil-cyber** [comment]
Hey @karthiksathishjeemain Done....removed the hardcoded addresses.

**Comment by santhil-cyber** [comment]
@pradeeban @karthiksathishjeemain Just a quick heads up ... since I was working on both branches in parallel, depending on which PR gets merged first between #235 and #237, there might be a small merge conflict in analytics-service/app/main.py where both add new endpoints and imports around the same lines.

**Comment by karthiksathishjeemain** [comment]
#235 is already approved, just waiting for it to get merged by @pradeeban . gimme some time to review ur changes for the current PR.

**Comment by pradeeban** [comment]
@santhil-cyber pls address the merge conflict.

**Comment by santhil-cyber** [comment]
Done @pradeeban 🫡

File: nextjs-frontend/.env.example [hunk 2]
```diff
@@ -7,3 +7,7 @@ NEXT_PUBLIC_JS_BACKEND_URL=http://localhost:3001
 # 32-byte encryption key for AES-256 encryption
 # Generate using: openssl rand -hex 32
 NEXT_PUBLIC_ENCRYPTION_KEY=your_32_byte_encryption_key_here
+
+# On-chain contract addresses
+NEXT_PUBLIC_CONTRACT_ADDRESS=0xYourDocumentStorageContractAddress
+NEXT_PUBLIC_ANALYTICS_REGISTRY_ADDRESS=0xYourAnalyticsRegistryContractAddress
```
Comment (karthiksathishjeemain) [inline nextjs-frontend/.env.example:13]: thanks for making the changes

**Review by karthiksathishjeemain — APPROVED** [review]
@santhil-cyber Thanks for storing the key inside .env file and resolving the merge conflicts. @pradeeban please merge this PR

**Comment by pradeeban** [comment]
Thanks, merged.
